tests/MonitorRetryTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace romanzipp\QueueMonitor\Tests;
+
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use romanzipp\QueueMonitor\Models\Monitor;
+use romanzipp\QueueMonitor\Tests\Support\MonitoredFailingJob;
+use romanzipp\QueueMonitor\Tests\Support\MonitoredJob;
+use romanzipp\QueueMonitor\Tests\TestCases\DatabaseTestCase;
+
+class MonitorRetryTest extends DatabaseTestCase
+{
+    public function setUp(): void
+    {
+        putenv('QUEUE_MONITOR_UI_ENABLED=true');
```

Comment:
To circumvent the env var, you can us the config helper function

```suggestion
        config(['queue-monitor.ui.ebabled' => true]);
```

Reply:
Unfortunately your suggestion won't work. Running `config()` (or any other framework logic) wouldn't work because the application is not booted yet. 
<img width="949" alt="application-not-booted" src="https://user-images.githubusercontent.com/79970817/236904862-5ec26a27-d0be-4478-8d04-a81b0e3f00d4.png">

`parent::setUp();` wil boot up the framework and will run the service provider. So setting `config(['queue-monitor.ui.enabled' => true]);` after the bootup wouldn't work either due to this line: https://github.com/romanzipp/Laravel-Queue-Monitor/blob/31a62b05ff142d22aae26f85311811198745ea9c/src/Providers/QueueMonitorProvider.php?plain=1#L49
<img width="978" alt="route-not-defined" src="https://user-images.githubusercontent.com/79970817/236905026-bd67b39e-d1ec-4240-ac6f-dc3a0005aaf9.png">

So the only suggestion I could think of, without changing the QueueMonitorProvider, was to add an env variable so we can change the config for route testing. That was the reason I introduced  'QUEUE_MONITOR_UI_ENABLED'. I understand your concern to not add an env at this point so I think a proper solution would be to add middleware. In the middleware I could check per GUI route if the related config is enabled and if not return a 404. Because the middleware (and not the booted provider) will check the configs the test wil work if we first boot, change the config and then do the request.

How do you think about the middleware solution instead of the env variable?